modules/openfga/openfga.go

```diff
@@ -0,0 +1,92 @@
+package openfga
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+// OpenFGAContainer represents the OpenFGA container type used in the module
+type OpenFGAContainer struct {
+	testcontainers.Container
+}
+
+// WithLogLevel returns a CustomizeRequestOption that sets the log level
+func WithLogLevel(level string) testcontainers.CustomizeRequestOption {
+	return testcontainers.WithEnv(map[string]string{"OPENFGA_LOG_LEVEL": level})
+}
+
+// WithPresharedAuthn returns a CustomizeRequestOption that sets the preshared authentication method
+// with the given secret
+func WithPresharedAuthn(secret string) testcontainers.CustomizeRequestOption {
+	return testcontainers.WithEnv(map[string]string{
+		"OPENFGA_AUTHN_METHOD":         "preshared",
+		"OPENFGA_AUTHN_PRESHARED_KEYS": secret,
+	})
+}
```

Comment:
This can be done with existing Testcontainers API without needing to add custom fn. Let's take into account that this can increase or become complex in the future for maintenance purposes.

Reply:
That's correct. I'll remove it

Reply:
✅ done in [412deaf](https://github.com/testcontainers/testcontainers-go/pull/2332/commits/412deaf6d04384fb5d135e2c99de11bdbeb48d8f)